pkg/extract/extract.go

```diff
@@ -145,17 +145,24 @@ func makeOptions(opts ...Option) (*options, error) {
 
 // findPath walks up the path, finding the longest match in the dirs map
 func findPath(dirs map[string]string, path string) (string, error) {
+	// Standardizing the string to a path to accommodate OS.
+	path = filepath.FromSlash(path)
+
 	if !strings.HasPrefix(path, ps) {
 		path = ps + path
 	}
 
 	for s := filepath.Dir(path); ; s = filepath.Dir(s) {
-		if d, ok := dirs[s]; ok {
+		// s needs to be converted back to unix style path to find the key
+		if d, ok := dirs[filepath.ToSlash(s)]; ok {
 			j := filepath.Clean(filepath.Join(d, strings.TrimPrefix(path, s)))
+
```

Comment:
I think you can do this without converting FromSlashes up front, and then converting back ToSlashes on every step up the path to the root. `d` (the destination mapping) should already be in host-native format. `path` (the current file) and `s` (the source mapping) are in unix format from the tarfile. All you should need to convert FromSlashes is the trimmed prefix, prior to joining it with the destination into `j`.

```suggestion
		if d, ok := dirs[filepath.ToSlash(s)]; ok {
			// normalize to platform-specific path format before joining with the mapped destination
			n := filepath.FromSlash(strings.TrimPrefix(path, s))
			j := filepath.Clean(filepath.Join(d, n))

```

Reply:
In my testing, this was required to make it work reliably on both Windows and Linux operating system with both Windows and Linux containers. If Windows only needed Windows container or Linux only needed Linux.

Reply:
When filepath.Clean is called it will convert it to windows os path separator, but the key is unix style and a string. So it needs to be converted.

Reply:
I'm confused then. The paths in the image are always Unix paths, regardless of whether they're Linux or Windows images, correct? So only the destination `d` (the map value) and the joined clean path `j` should ever have to deal with platform-specific paths.

Reply:
We need to step through it together tomorrow. Any time a filepath package function touches a string it converts it to the host OS path separator. So the unix paths get converted to Windows paths. We then compare those filepath generated strings to "raw" strings that are still unix style paths. We have to always ensure that the resulting strings are compared with the same path separators.

Reply:
I would like to move forward with this so as to unblock the windows folks.

Reply:
that's fine, I'll just fix it when I rebase #5.

Reply:
The correct fix is to use `path` which always uses unix path separators for tarfile paths, instead of filepath which wants to use os-specific separators. If you do that then the converting back and forth is unnecessary.